views/lti/editorMethods.jade

```diff
@@ -10,34 +10,46 @@ script
               removePlugins: 'resize',
               sharedSpaces: { top: 'ckeditor-toolbar' },
               on: {
-                key: () => {
-                  const bindedEditorId = editor.id.replace('canvas-', '')
-
-                  setTimeout(() => CKEDITOR.instances[bindedEditorId].setData(
-                    CKEDITOR.instances[editor.id].getData()
-                  ), 0)
-                }
+                key: canvasEditorOnKeyHandler
               }
             })
           } else {
             CKEDITOR.replace(editor.id, {
               on: {
-                key: () => {
-                  const bindedEditorId = `canvas-${editor.id}`
-
-                  // if the deliverable has no canvas there'll be no binded editors
-                  if (CKEDITOR.instances[bindedEditorId]) {
-                    setTimeout(() => CKEDITOR.instances[bindedEditorId].setData(
-                      CKEDITOR.instances[editor.id].getData()
-                    ), 0)
-                  }
-                }
+                key: editorOnKeyHandler
               }
             })
           }
         })
       }
 
+      function editorOnKeyHandler() {
+        const editorId = this.name
+        const bindedEditorId = `canvas-${this.name}`
+
+        // if the deliverable has no canvas there'll be no binded editors
+        if (CKEDITOR.instances[bindedEditorId] && !this.syncing) {
+          this.syncing = true
+          setTimeout(() => {
```

Comment:
Remove lambda and instead of this.syncing, save setTimeout reference and use that as semaphore.